src/pilot_weapon.c

```diff
@@ -1164,6 +1163,9 @@ static int pilot_shootWeapon( Pilot* p, PilotOutfitSlot* w, double time )
    /* Reset timer. */
    w->timer += rate_mod * outfit_delay( w->outfit );
 
+   /* Shooting prevents reloading. Reset reload timer. */
+   w->rtimer = 0;
```

Comment:
Should shooting prevent reloading? Not necessarily certain, because it will encourage the player to sort of min-max stuff. Might be better to just let it reload while shooting and adjust the outfit parameters to take that into account.

Reply:
The idea I had isn't so much to limit reload rate as to prevent it from happening while you're using it, if that makes sense? Basically the goal was to create a dichotomy between "in use" and "being replenished" for any given launcher.

It can be removed if you want. I like the effect personally, though. Basically I was trying to make it so that you're encouraged to use missiles in concentrated volleys.

Reply:
One thing I think is a must if that's removed, the faster reload rates for Mace Rockets and that other dumbfire rocket would need to be increased to compensate. Maybe I could kick them all to a standard 60 seconds or something.

Reply:
Testing some changes to this effect now.

Reply:
> Basically the goal was to create a dichotomy between "in use" and "being replenished" for any given launcher.

If we keep this behavior, it might be nice if the current state was indicated in the UI in some way. Certainly not critical though.

I like the idea of not reloading while shooting. But I'm not really sure which will would out better in practice.

Reply:
Admittedly it doesn't make that big of a difference if the reload late is very very slow anyway. But at that point I think it more or less just encourages active cooling.

Reply:
OK, so I pushed a change getting rid of the "no reload while shooting" restriction (d01b1d6) and then a reversion to that (52ec9e8).

Honestly, technically any way would allow for any possibility to work right, it's just a question of what you need to do to achieve that. I mostly like this way better:

1. Because it's easier to design and balance that way (rather than having to go through math to find out what the true ammo capacity is after regen the numbers are pretty straightforward); and
2. Because it encourages using missiles in a different way (missile barrages rather than ammo conservation; the latter is already heavily encouraged for regular weapons by the heat system).

Reply:
I did some thinking about this and have decided that honestly, it doesn't really matter all that much. Calculations show that the effective max ammo of nearly all rockets won't be increased at all, and of those whose effective max ammo does increase, it's not that much (e.g. with a reload wait of 6 Mace Rockets have I think around 4 extra missiles in a full volley).